packages/gatsby-image/src/index.js

```diff
@@ -239,24 +241,51 @@ const listenToIntersections = (el, cb) => {
 }
 
 const noscriptImg = props => {
-  // Check if prop exists before adding each attribute to the string output below to prevent
-  // HTML validation issues caused by empty values like width="" and height=""
-  const src = props.src ? `src="${props.src}" ` : `src="" ` // required attribute
-  const sizes = props.sizes ? `sizes="${props.sizes}" ` : ``
-  const srcSet = props.srcSet ? `srcset="${props.srcSet}" ` : ``
-  const title = props.title ? `title="${props.title}" ` : ``
-  const alt = props.alt ? `alt="${props.alt}" ` : `alt="" ` // required attribute
-  const width = props.width ? `width="${props.width}" ` : ``
-  const height = props.height ? `height="${props.height}" ` : ``
-  const crossOrigin = props.crossOrigin
-    ? `crossorigin="${props.crossOrigin}" `
-    : ``
-  const loading = props.loading ? `loading="${props.loading}" ` : ``
-  const draggable = props.draggable ? `draggable="${props.draggable}" ` : ``
+  const {
+    src = ``,
+    alt = ``,
+    sizes,
+    srcSet,
+    title,
+    width,
+    height,
+    crossOrigin,
+    loading,
+    draggable,
+  } = props
 
+  const imageProps = {
+    src,
+    alt,
+    sizes,
+    srcSet,
+    title,
+    width,
+    height,
+    crossOrigin,
+    loading,
+    draggable,
+
+    style: {
+      position: `absolute`,
+      top: 0,
+      left: 0,
+      opacity: 1,
+      width: `100%`,
+      height: `100%`,
+      objectFit: `cover`,
+      objectPosition: `center`,
+      ...props.imgStyle,
+    },
+  }
   const sources = generateNoscriptSources(props.imageVariants)
 
-  return `<picture>${sources}<img ${loading}${width}${height}${sizes}${srcSet}${src}${alt}${title}${crossOrigin}${draggable}style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/></picture>`
+  return ReactDOMServer.renderToString(
+    <picture>
+      {sources}
+      <img {...imageProps} />
+    </picture>
+  )
```

Comment:
Nice workaround :)

I guess this would work until v3, but maybe add a TODO comment about that, like:

```
// TODO: In v3, minimum required React will be >=16.6, 'dangerouslySetInnerHTML' and 'ReactDOMServer' workarounds will no longer be necessary and can be dropped
```

Additionally, it would make sense to move the `<noscript />` element and it's `dangerouslySetInnerHTML` prop into this component and return that after providing it the string? Should isolate/group the issue better that way for the v3 refactor.